src/sentry/static/sentry/app/components/globalSdkSuggestions.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import {promptsCheck, promptsUpdate} from 'app/actionCreators/prompts';
+import SidebarPanelActions from 'app/actions/sidebarPanelActions';
+import {Client} from 'app/api';
+import Alert from 'app/components/alert';
+import {IconUpgrade} from 'app/icons';
+import {t, tct} from 'app/locale';
+import space from 'app/styles/space';
+import {
+  GlobalSelection,
+  Organization,
+  ProjectSdkUpdates,
+  SDKUpdatesSuggestion,
+} from 'app/types';
+import {snoozedDays} from 'app/utils/promptsActivity';
+import withApi from 'app/utils/withApi';
+import withGlobalSelection from 'app/utils/withGlobalSelection';
+import withOrganization from 'app/utils/withOrganization';
+import withSdkUpdates from 'app/utils/withSdkUpdates';
+
+import {SidebarPanelKey} from './sidebar/types';
+import Button from './button';
+
+type Props = {
+  api: Client;
+  organization: Organization;
+  sdkUpdates?: ProjectSdkUpdates[] | null;
+  selection?: GlobalSelection;
+};
+
+type State = {
+  isDismissed: boolean | null;
+};
+
+const flattenSuggestions = (list: ProjectSdkUpdates[]) =>
+  list.reduce<SDKUpdatesSuggestion[]>(
+    (suggestions, sdk) => [...suggestions, ...sdk.suggestions],
+    []
+  );
+
+class InnerGlobalSdkSuggestions extends React.Component<Props, State> {
+  state: State = {
+    isDismissed: null,
+  };
+
+  componentDidMount() {
+    this.promptsCheck();
+  }
+
+  async promptsCheck() {
+    const {api, organization} = this.props;
+
+    const prompt = await promptsCheck(api, {
+      organizationId: organization.id,
+      feature: 'sdk_updates',
+    });
+
+    this.setState({
+      isDismissed: !prompt?.snoozedTime ? false : snoozedDays(prompt?.snoozedTime) < 14,
+    });
+  }
+
+  dismissPrompt = () => {
+    const {api, organization} = this.props;
+    promptsUpdate(api, {
+      organizationId: organization.id,
+      feature: 'sdk_updates',
+      status: 'snoozed',
+    });
+
+    this.setState({isDismissed: true});
+  };
+
+  render() {
+    const {selection, sdkUpdates} = this.props;
+    const {isDismissed} = this.state;
+
+    if (!sdkUpdates || isDismissed === null || isDismissed) {
+      return null;
+    }
+
+    const projectSpecificUpdates =
+      selection?.projects.length === 0
```

Comment:
I think this check here needs to handle `My Projects` vs `All Projects`

- `[]` should be checking for `My Projects`
- `[-1]` should be checking for `All Projects`

Reply:
Currently this just supports 'My projects' or or a subset of this set of projects.

The reason is, because the withSdkUpdates is a global HoC, it requests the sdk-updates once, for My Projects, and caches that with the organization-id as the key. If we were using globalSelection to make the sdk-updates request, we would need to associate the cache key to those projects, which might just be additional complexity for little benifit.